### ISSUE TITLE
Update `rasa-sdk` dependencies to align with those of `rasa`

### ISSUE THREAD

**Issue by ancalita** [RasaHQ/rasa-sdk#592]
**Description of Problem**:

Installing editable version of `rasa-sdk` after having installed editable version of `rasa` in the same venv causes the bot to crash when running `rasa shell`, full stack trace below:

```
Bot loaded. Type a message and press enter (use '/stop' to exit): 
2021-11-11 13:26:28 ERROR    asyncio  - Task exception was never retrieved
future: <Task finished name='Task-6' coro=<SignalRouter._dispatch() done, defined at /Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py:102> exception=RuntimeError('this event loop is already running.') created at /Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py:165>
source_traceback: Object created at (most recent call last):
  File "/Users/ancalita/rasa-projects/rasa-oss-09/bin/rasa", line 5, in <module>
    main()
  File "/Users/ancalita/rasa-projects/rasa/rasa/__main__.py", line 121, in main
    cmdline_arguments.func(cmdline_arguments)
  File "/Users/ancalita/rasa-projects/rasa/rasa/cli/shell.py", line 125, in shell
    rasa.cli.run.run(args)
  File "/Users/ancalita/rasa-projects/rasa/rasa/cli/run.py", line 126, in run
    rasa.run(**vars(args))
  File "/Users/ancalita/rasa-projects/rasa/rasa/api.py", line 56, in run
    rasa.core.run.serve_application(
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/run.py", line 221, in serve_application
    app.run(
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/app.py", line 1051, in run
    serve_single(server_settings)
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/server/runners.py", line 199, in serve_single
    serve(**server_settings)
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/server/runners.py", line 146, in serve
    loop.run_until_complete(app._server_event("init", "after"))
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/app.py", line 1496, in _server_event
    await self.dispatch(
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py", line 163, in dispatch
    return await dispatch
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py", line 133, in _dispatch
    retval = await maybe_coroutine
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/app.py", line 1382, in _listener
    await maybe_coro
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/app.py", line 1368, in dispatch_delayed_tasks
    await app.dispatch(name, context={"app": app, "loop": loop})
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py", line 165, in dispatch
    task = asyncio.get_running_loop().create_task(dispatch)
Traceback (most recent call last):
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/signals.py", line 133, in _dispatch
    retval = await maybe_coroutine
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/sanic/app.py", line 1374, in run_delayed_task
    await prepped
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/run.py", line 134, in run_cmdline_io
    await console.record_messages(
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/channels/console.py", line 175, in record_messages
    text = _get_user_input(previous_response)
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/channels/console.py", line 108, in _get_user_input
    response = questionary.text(
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/questionary/question.py", line 50, in ask
    return self.unsafe_ask(patch_stdout)
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/questionary/question.py", line 64, in unsafe_ask
    return self.application.run()
  File "/Users/ancalita/rasa-projects/rasa-oss-09/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 890, in run
    return loop.run_until_complete(
  File "uvloop/loop.pyx", line 1450, in uvloop.loop.Loop.run_until_complete
  File "uvloop/loop.pyx", line 1443, in uvloop.loop.Loop.run_until_complete
  File "uvloop/loop.pyx", line 1351, in uvloop.loop.Loop.run_forever
  File "uvloop/loop.pyx", line 480, in uvloop.loop.Loop._run
```

This [forum thread](https://forum.rasa.com/t/error-this-event-loop-is-already-running/24017/9) seems to pin the cause of this issue to the version of `questionary` which I noticed `rasa-sdk` downgrades from `1.10` to `1.8` when running `make install`.

**Definition of Done**:
- [ ] `questionary` version in `rasa-sdk` is pinned to the higher version as in rasa-oss
- [ ] double-check all dependencies are pinned to the higher version pinned in rasa-oss
- [ ] you are able to run `rasa shell` without issues when both projects are installed as editable in the same venv

**Comment by carlad** [comment]
I'm unable to replicate this issue after following the steps provided in the same virtualenv:
- `make install` rasa
- `make install` rasa-sdk
- `rasa init` a new project
- `rasa shell`

**Comment by carlad** [comment]
The diff in `2.8.x`

```
rasa                                                        | rasa-sdk
aiofiles                 0.7.0				      |	aiofiles                 0.8.0
azure-core               1.20.1				      |	azure-core               1.21.1
boto3                    1.20.21				  | boto3                    1.20.6
botocore                 1.23.21				  | botocore                 1.23.6
colorhash                1.0.3				      |	colorhash                1.0.4
coverage                 6.1.2				      |	coverage                 6.2
dask                     2021.12.0			      |	dask                     2021.11.2
decorator                5.1.0                    | decorator                4.4.2
fakeredis                1.6.1				      |	fakeredis                1.7.0
filelock                 3.4.0				      |	filelock                 3.4.2
GitPython                3.1.20				      |	GitPython                3.1.24
google-cloud-storage     1.42.3				      |	google-cloud-storage     1.43.0
grpcio                   1.41.1				      |	grpcio                   1.42.0
h11                      0.9.0				      |	h11                      0.12.0
httpcore                 0.11.1				      |	httpcore                 0.13.7
httpx                    0.11.1				      |	httpx                    0.18.2
importlib-metadata       4.10.0				      |	importlib-metadata       4.8.2
kiwisolver               1.3.1				      |	kiwisolver               1.3.2
Markdown                 3.3.5				      |	Markdown                 3.3.6
moto                     2.0.11				      |	moto                     2.2.17
msgpack                  1.0.2				      |	msgpack                  1.0.3
multidict                4.7.6				      |	multidict                5.2.0
networkx                 2.5.1				      |	networkx                 2.6.3
pbr                      5.7.0				      |	pbr                      5.8.0
pydantic                 1.7.4				      |	pydantic                 1.8.2
PyJWT                    2.0.0                    |	PyJWT                    2.1.0
pytest-sanic             1.6.2				      |	pytest-sanic             1.9.1
questionary              1.9.0				      |	questionary              1.10.0
rasa                     3.0.0				      |	rasa                     3.0.4
rasa-sdk                 2.8.2				      |	rasa-sdk                 3.0.2
regex                    2021.7.6			      |	regex                    2021.8.28
rsa                      4.7.2				      |	rsa                      4.8
sanic                    20.3.0				      |	sanic                    21.9.3
Sanic-Cors               0.10.0.post3			  |	Sanic-Cors               1.0.1
sanic-jwt                1.6.0   			      |	sanic-jwt                1.7.0
scipy                    1.5.4				      |	scipy                    1.7.3
sentry-sdk               1.2.0				      |	sentry-sdk               1.3.1
setuptools               56.0.0				      |	setuptools               60.2.0
snowballstemmer          2.1.0				      |	snowballstemmer          2.2.0
spacy                    3.2.0				      |	spacy                    3.2.1
tensorboard              2.6.0				      |	tensorboard              2.7.0
tensorflow               2.6.2				      |	tensorflow               2.6.1
terminaltables           3.1.0				      |	terminaltables           3.1.7
typeguard                2.13.0				      |	typeguard                2.13.2
types-pytz               2021.3.0			      |	types-pytz               2021.3.1
types-requests           2.26.0				      |	types-requests           2.26.1
types-setuptools         57.4.2				      |	types-setuptools         57.4.4
ujson                    4.2.0				      |	ujson                    4.3.0
wasabi                   0.8.2				      |	wasabi                   0.9.0
websockets               8.0.2				      |	websockets               10.1
wheel                    0.37.0				      |	wheel                    0.37.1
zipp                     3.7.0				      |	zipp                     3.6.0
```

**Comment by carlad** [comment]
diff `3.0.x` versions

rasa                                              | rasa-sdk
--------------------------------------------------|----------------------------------
aiofiles                 0.7.0                    | aiofiles                 0.8.0
azure-core               1.20.1                   | azure-core               1.21.1
boto3                    1.20.21                  | boto3                    1.20.6
botocore                 1.23.6                   | botocore                 1.23.21
colorhash                1.0.3                    | colorhash                1.0.4
coverage                 6.1.2                    | coverage                 6.2
dask                     2021.12.0                | dask                     2021.11.2
decorator                4.4.2                    | decorator                5.1.0
fakeredis                1.6.1                    | fakeredis                1.7.0
filelock                 3.4.0                    | filelock                 3.4.2
GitPython                3.1.20                   | GitPython                3.1.24
google-cloud-storage     1.42.3                   | google-cloud-storage     1.43.0
grpcio                   1.41.1                   | grpcio                   1.42.0
h11                      0.9.0                    | h11                      0.12.0
httpcore                 0.11.1                   | httpcore                 0.13.7
httpx                    0.11.1                   | httpx                    0.18.2
importlib-metadata       4.10.0                   | importlib-metadata       4.8.2
kiwisolver               1.3.1                    | kiwisolver               1.3.2
Markdown                 3.3.5                    | Markdown                 3.3.6
moto                     2.0.11                   | moto                     2.2.17
msgpack                  1.0.2                    | msgpack                  1.0.3
multidict                4.7.6                    | multidict                5.2.0
networkx                 2.5.1                    | networkx                 2.6.3
pbr                      5.7.0                    | pbr                      5.8.0
pydantic                 1.7.4                    | pydantic                 1.8.2
PyJWT                    2.0.0                    | PyJWT                    2.1.0
pytest-sanic             1.6.2                    | pytest-sanic             1.9.1
questionary              1.9.0                    | questionary              1.10.0
rasa                     3.0.0                    | rasa                     3.0.4
rasa-sdk                 2.8.2                    | rasa-sdk                 3.0.2
regex                    2021.7.6                 | regex                    2021.8.28
rsa                      4.7.2                    | rsa                      4.8
sanic                    20.3.0                   | sanic                    21.9.3
Sanic-Cors               0.10.0.post3             | Sanic-Cors               1.0.1
sanic-jwt                1.6.0                    | sanic-jwt                1.7.0
scipy                    1.5.4                    | scipy                    1.7.3
sentry-sdk               1.2.0                    | sentry-sdk               1.3.1
setuptools               56.0.0                   | setuptools               60.2.0
snowballstemmer          2.1.0                    | snowballstemmer          2.2.0
spacy                    3.2.0                    | spacy                    3.2.1
tensorboard              2.6.0                    | tensorboard              2.7.0
tensorflow               2.6.2                    | tensorflow               2.6.1
terminaltables           3.1.0                    | terminaltables           3.1.7
typeguard                2.13.0                   | typeguard                2.13.2
types-pytz               2021.3.0                 | types-pytz               2021.3.1
types-requests           2.26.0                   | types-requests           2.26.1
types-setuptools         57.4.2                   | types-setuptools         57.4.4
ujson                    4.2.0                    | ujson                    4.3.0
wasabi                   0.8.2                    | wasabi                   0.9.0
websockets               8.0.2                    | websockets               10.1
wheel                    0.37.0                   | wheel                    0.37.1
zipp                     3.7.0                    | zipp                     3.6.0

**Comment by carlad** [comment]
@ancalita 
> Related to the dependencies' diff: are you comparing 3.0.x or 2.8.x?
I'm doing both, though it seems the dependencies are similar...

**Comment by ancalita** [comment]
@carlad I'm seeing this issue again just now in a fresh venv with rasa and rasa-sdk installed via `make install` 🙈 
`questionary` is version `1.10`, and the project I'm using has a custom action defined and runs the actions server in a separate tab too.
When you have some time, could you please check with a similar project to see if `rasa shell` crashes for you too?

```
(oss-0122) ➜  required_slots_formvalidationaction rasa shell
2022-01-24 16:19:36 INFO     root  - Connecting to channel 'cmdline' which was specified by the '--connector' argument. Any other channels will be ignored. To connect to all given channels, omit the '--connector' argument.
2022-01-24 16:19:36 INFO     root  - Starting Rasa server on http://0.0.0.0:5005
2022-01-24 16:19:37 INFO     rasa.core.processor  - Loading model models/20220124-161441-humid-duplet.tar.gz...
2022-01-24 16:19:50 WARNING  rasa.shared.utils.common  - The UnexpecTED Intent Policy is currently experimental and might change or be removed in the future 🔬 Please share your feedback on it in the forum (https://forum.rasa.com) to help us make this feature ready for production.
2022-01-24 16:19:56 INFO     root  - Rasa server is up and running.
Bot loaded. Type a message and press enter (use '/stop' to exit): 
2022-01-24 16:19:57 ERROR    asyncio  - Task exception was never retrieved
future: <Task finished name='Task-6' coro=<SignalRouter._dispatch() done, defined at /Users/ancalita/rasa-projects/oss-0122/lib/python3.8/site-packages/sanic/signals.py:102> exception=RuntimeError('this event loop is already running.')>
Traceback (most recent call last):
  File "/Users/ancalita/rasa-projects/oss-0122/lib/python3.8/site-packages/sanic/signals.py", line 133, in _dispatch
    retval = await maybe_coroutine
  File "/Users/ancalita/rasa-projects/oss-0122/lib/python3.8/site-packages/sanic/app.py", line 1374, in run_delayed_task
    await prepped
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/run.py", line 134, in run_cmdline_io
    await console.record_messages(
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/channels/console.py", line 175, in record_messages
    text = _get_user_input(previous_response)
  File "/Users/ancalita/rasa-projects/rasa/rasa/core/channels/console.py", line 108, in _get_user_input
    response = questionary.text(
  File "/Users/ancalita/rasa-projects/oss-0122/lib/python3.8/site-packages/questionary/question.py", line 70, in ask
    return self.unsafe_ask(patch_stdout)
  File "/Users/ancalita/rasa-projects/oss-0122/lib/python3.8/site-packages/questionary/question.py", line 92, in unsafe_ask
    return self.application.run()
  File "/Users/ancalita/rasa-projects/oss-0122/lib/python3.8/site-packages/prompt_toolkit/application/application.py", line 890, in run
    return loop.run_until_complete(
  File "uvloop/loop.pyx", line 1450, in uvloop.loop.Loop.run_until_complete
  File "uvloop/loop.pyx", line 1443, in uvloop.loop.Loop.run_until_complete
  File "uvloop/loop.pyx", line 1351, in uvloop.loop.Loop.run_forever
  File "uvloop/loop.pyx", line 480, in uvloop.loop.Loop._run
RuntimeError: this event loop is already running.
```

**Comment by ancalita** [comment]
Could it be related to this issue: https://github.com/RasaHQ/rasa/issues/10632?

**Comment by carlad** [comment]
So, without `make install`ing anything,  I wasn't able to replicate this working inside `rasa-x-demo` directory. 
I was able to run `rasa shell` with no issues on either a `rasa` `2.8.x` or `3.0.x` virtualenv. 
```
Rasa Version      :         2.8.15
Minimum Compatible Version: 2.8.9
Rasa SDK Version  :         2.8.2
Rasa X Version    :         1.0.1
Python Version    :         3.7.12
```

With `rasa` `3.0.x` I'm also able to execute `rasa shell` ok.
```
Rasa Version      :         3.0.0
Minimum Compatible Version: 3.0.0
Rasa SDK Version  :         3.0.3
Rasa X Version    :         1.0.1
Python Version    :         3.8.6
```

So I ran `make install` in the `rasa-sdk` directory. And then running `rasa shell` saw similar issues on `2.8.x` and `3.0.x`

At first I thought it was the `questionary` library version. There were several posts online from last year stating a rollback to v1.4.x fixed the issue. _However_ after mucking around with different `questionary` versions, settling finally on `v1.8.1`, I noticed that any time I ran `make install` in the `rasa` directory _last_, then ran `rasa shell`, it would launch fine. But anytime I ran `make install` in the `rasa-sdk` directory, _then_ rans `rasa shell`, it would produce this error.

So I'm thinking it's not about dependency conflicts, but about a loop process in the `make install` task that doesn't end/close properly, but somehow gets reset when `rasa` is `make install`ed .

**Comment by carlad** [comment]
So it turns out the issue was with the version of `prompt-toolkit` we were installing. Previously we had the version tied to `'^2.0'` in the `.toml` file, but I must have changed it to a range, which ended up installing the latest version (3.something).
Rolling back to `'^2.0'` solves the problem.

**Comment by carlad** [comment]
The fix now released in rasa 3.0.6 

https://github.com/RasaHQ/rasa/releases/tag/3.0.6

and rasa-sdk 3.0.4

https://github.com/RasaHQ/rasa-sdk/blob/3.0.4/CHANGELOG.mdx
